### DISCUSSION TITLE
Long Term Usage

### DISCUSSION THREAD

**Discussion by LaserEyess** [kanidm/kanidm#1477] [Q&A] [answered]
Let me preface this by saying that I do not mean for this to be rude, nor do I mean for this to be an assertion of "customer service" responsibility/liability from maintainers. 

My question is: what is the backing for this project and do you see it being maintainable in terms of keeping up with the ever-changing landscape of web technologies and security pitfalls? This software, from what I have seen so far, is incredible and fills a niche that I think many will find useful. It is also very security and protocol sensitive, and will require continuous maintenance even after being feature complete. Usually, projects of this magnitude and this much scope have some sort of backing in the form of commercial support, e.g. sponsored by a large company or as a company that sells enterprise licenses. I don't see much evidence for either of those from this project. 

I am testing this for myself as a solution for SSO for services for close friends and family who don't have the skills to host their own files, email, etc. So I want to use this long term, and I am willing to contribute to this project, but I'm really curious if you can share if there's a Big Backer(tm) or some other entity that can work to make sure this project has the development resources it needs.

**Comment by Firstyear** [comment, marked as answer]
I think this concern is very justified and reasonable. I don't think it's a criticism at all and you're right to want to check about sustainability of the project.

So I think there are a few dimensions to this:

* Are the developers committed to this project long term?
* Are the developers currently funded in some manner to have the time to work on this to maintain and improve the project?
* How/where is that funding coming from?
* Will that funding continue?
* Is there a risk of this project just "going away".

The project has been developed since 2018 and has attracted others. Both myself and @yaleman are in this for the long haul, and I have no plans to stop improving Kanidm! We both use it personally and have vested interests in it's sustainability and security.

There are multiple sources and methods of funding. I am an employee at SUSE and they have been supporting me to work on Kanidm in my work time and free time. @yaleman can choose if he wishes to disclose his employer and situation, but he certainly contributes a great deal. We have also been a GSOC mentoring project multiple times and those students have continued to be a part of the project in different capacities. In the (unlikely) absence of SUSE for myself, I have already discussed alternate funding plans with others to continue my work on this project. 

As for "how will that change in the future" that is one that's harder to discuss. Employees are often not at liberty to discuss business plans, but it is my view that SUSE want this project to succeed in the open. I have a positive outlook on this matter but I won't say more. 

Finally, all copyright of the project still belongs to the contributors who committed it and is not property of a company, so this project will stay in the open. 

I am very confident that Kanidm will be around for a long time given the current state of our community, it's growing contributors and user base.

**Comment by yaleman** [comment]
My employer's not supporting it in any way currently other than paying me to do my day-job, and being chill when I rant on about Rust and Webauthn 😄  I'll keep poking this crab down the path when I have new ideas and spare brain cells, or helping the community do their thing.

I don't see me going anywhere, either way. 

Nor is OAuth2, LDAP, RADIUS, Webauthn or any of the other tech that Kanidm supports, if you're running it now, without *epic* changes in the infrastructure of the web, they're going to stay working for a long time.

**Comment by LaserEyess** [comment]
Thanks guys, I appreciate the openness. I'm stuck between things that support LDAP and not oauth/oidc, and things that support oauth2/oidc and don't support LDAP. This project is an incredible bridge for that, so I'm definitely going to start using it.
